setup.cfg

```diff
@@ -16,6 +16,7 @@
 
 [options.entry_points]
 cylc.pre_configure = 
-    rose = cylc.rose.rose:get_rose_vars
+    rose = cylc.rose.entry_points:get_rose_vars
 cylc.post_install =
-    rose = cylc.rose.rose:rose_fileinstall
+    rose_opts = cylc.rose.entry_points:record_cylc_install_options
+    rose_files = cylc.rose.entry_points:rose_fileinstall
```

Comment:
The ability to use environment variables to control file installation has been broken again. Here's my test suite:

```
$ tree
.
├── base
├── flow.cylc
├── opt
│   └── rose-suite-x.conf
├── rose-suite.conf
└── x
```

```console
$ cat rose-suite.conf 
[env]
FILENAME=base

[file:myfile]
source=localhost:$PWD/$FILENAME
```

```console
$ cat opt/rose-suite-x.conf 
[env]
FILENAME=x
```

This should install a file called `myfile` which will either contain the contents of file `base` or file `x` depending on whether the `x` optional config is specified on the CLI.

```console
$ cylc install
...
PluginError: Error in plugin cylc.post_install.rose_files
file:myfile=source: FILENAME: unbound variable
```

This seems to be repeatable, it might be because the plugins are not necessarily called in the order that they are listed in the `setup.cfg`. I think it would be safer to group these two into a single entry point in order to ensure that they will always get run in this order, e.g. with a simple dumb wrapper like so:

```python
def whatever(*args, **kwargs):
    record_cylc_install_options(*args, **kwargs)
    rose_fileinstall(*args, **kwargs)
```

Reply:
>  it might be because the plugins are not necessarily called in the order that they are listed

The plugin which _should_ export `$FILENAME` to the environment is the pre-install plugin, so it should _always_ be run before the other two. So I don't think this is likely. More likely, somehow is environment issues:
Running `FILENAME=x cylc install` works. I wonder if the entry points run in their own envs.

Reply:
I'm not aware of any gaurentee that entry points will be called in the order they are specified in `setip.cfg` so yep, good to try a wrapper.